common/model-views.cpp

```diff
@@ -4183,6 +4234,133 @@ namespace rs2
+                        device_calibration dev_cal(dev);
+                        // Register AC status change callback
+                        dev_cal.register_calibration_change_callback([&](rs2_calibration_status cal_stts)
+                        {
+                            cah_model.calib_stts = cal_stts;
+                            if (cah_model.calib_stts == RS2_CALIBRATION_STARTED || cah_model.calib_stts == RS2_CALIBRATION_SPECIAL_FRAME)
+                            {
+                                cah_model.cah_state = camera_accuracy_health_model::model_state_type::PROCESS_MODAL;
+                            }
+                        });
+
+                    }
+                    else
+                    {
+                        std::stringstream ss;
+                        ss << "Can not trigger Camera Accuracy Health, required streams not active";
+                        viewer.not_model->add_log(ss.str());
+                    }
+                }
+                else
+                {
+                    keep_showing = false;
+                }
+            }
+        }
+        break;
+
+        case camera_accuracy_health_model::model_state_type::PROCESS_MODAL:
+        {
+            bool process_finished(cah_model.calib_stts == RS2_CALIBRATION_SUCCESSFUL || cah_model.calib_stts == RS2_CALIBRATION_FAILED || cah_model.calib_stts == RS2_CALIBRATION_NOT_NEEDED);
+            
+            static std::map<rs2_calibration_status, std::string> status_map{
+                {RS2_CALIBRATION_SPECIAL_FRAME  , "In Progress" },
+                {RS2_CALIBRATION_STARTED        , "In Progress" },
+                {RS2_CALIBRATION_NOT_NEEDED     , "Ended" },
+                {RS2_CALIBRATION_SUCCESSFUL     , "Ended Successfully" },
+                {RS2_CALIBRATION_RETRY          , "In Progress" },
+                {RS2_CALIBRATION_FAILED         , "Ended With Failure" },
+                {RS2_CALIBRATION_SCENE_INVALID  , "In Progress" },
+                {RS2_CALIBRATION_BAD_RESULT     , "In Progress" } };
+
+
+            if (process_model("Camera Accuracy Health", "Camera Accuracy Health is In progress...  this may take a while.", status_map[cah_model.calib_stts], process_finished, window))
+            {
+                keep_showing = false;
+            }
+        }
+        break;
+        default:
+            break;
+
+        }
+
+        //reset internal elements
+        if (!keep_showing)
+        {
+            cah_model.cah_state = camera_accuracy_health_model::model_state_type::TRIGGER_MODAL;
+            cah_model.calib_stts = RS2_CALIBRATION_RETRY;
+        }
+        return keep_showing;
+        ;
+    }
+
+    bool device_model::prompt_reset_camera_accuracy_health(ux_window& window, std::string& error_message)
+    {
+        bool keep_showing = true;
+        bool yes_was_chosen = false;
+
+        std::string message_text("Would you like to reset Camera Accuracy Health ");
```

Comment:
Missing question mark?

Reply:
Need to change:
This will reset the camera settings to their factory-calibrated state. You will lose any improvements made with Camera Accuracy Health. Are you sure?

Reply:
1. The question mark was a built in icon, now it's not. I will update to your suggestion.